ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixParticipant.java

```diff
@@ -155,14 +155,34 @@ public boolean setReplicaSealedState(ReplicaId replicaId, ReplicaSealStatus repl
     synchronized (helixAdministrationLock) {
       DataNodeConfig config = getDataNodeConfig();
       String partitionId = replicaId.getPartitionId().toPathString();
-      boolean success = true;
-      // TODO Handle partial seal case.
-      if (replicaSealStatus != ReplicaSealStatus.SEALED && config.getSealedReplicas().remove(partitionId)) {
-        logger.trace("Removing the partition {} from sealedReplicas list", partitionId);
-        success = dataNodeConfigSource.set(config);
-      } else if (replicaSealStatus == ReplicaSealStatus.SEALED && config.getSealedReplicas().add(partitionId)) {
-        logger.trace("Adding the partition {} to sealedReplicas list", partitionId);
-        success = dataNodeConfigSource.set(config);
+      boolean success = false;
+      boolean configChanged;
+      switch (replicaSealStatus) {
+        case NOT_SEALED:
+          configChanged = config.getSealedReplicas().remove(partitionId);
+          configChanged = config.getPartiallySealedReplicas().remove(partitionId) || configChanged;
+          if (configChanged) {
+            logger.trace("Removing partition {} from sealed lists to mark it as NOT_SEALED", partitionId);
+            success = dataNodeConfigSource.set(config);
+          }
+          break;
+        case PARTIALLY_SEALED:
```

Comment:
Just double check. At this moment, no code will call HelixParticipant.setReplicaSealedState, right?

Reply:
There is code that calls `HelixParticipant.setReplicaSealedState` (see ReplicaStatusDelegate.java). But note that it's never called with replicaSealStatus argument set to `ReplicaSealStatus.Partially_Sealed`.